symengine/dense_matrix.cpp

```diff
@@ -14,11 +14,6 @@ DenseMatrix::DenseMatrix()
 {
 }
 
-DenseMatrix::DenseMatrix(const DenseMatrix &x)
-    : m_(x.m_), row_(x.row_), col_(x.col_)
-{
-}
```

Comment:
+1

I think for speed reasons, all these constructors should be in the header file so that they can be inlined. 

I can see that you made them `default`, so that's even better.